core/domain/caching_services.py

```diff
@@ -52,6 +53,11 @@
         platform_parameter_domain.PlatformParameter
     ]
 
+
+if MYPY: # pragma: no cover
+    from mypy_imports import memory_cache_services
+
+
```

Comment:
I think you can move `from mypy_imports import memory_cache_services` under the `if TYPE_CHECKING`